### DISCUSSION TITLE
do you save keypoints through frames ?

### DISCUSSION THREAD

**Discussion by an99990** [stella-cv/stella_vslam#319] [Q&A] [answered]
In your paper, it is mentioned that you do tracking, however I am unable to find where in the source code you keep track of the features detected. 
Is it possible to refer me to where the tracking is happening ? how do you keep the features over time ?

Thank you

**Comment by ymd-stella** [comment, marked as answer]
> In your paper

There is a misunderstanding. The paper is a copyrighted work of AIST. I am not a member of AIST, hence the name change to OpenVSLAM. Hence the name change from OpenVSLAM to stella_vslam, see https://github.com/stella-cv/stella_vslam/discussions/267.

> it is mentioned that you do tracking, however I am unable to find where in the source code you keep track of the features detected.
> Is it possible to refer me to where the tracking is happening ? how do you keep the features over time ?

Feature points are not directly tracked. Landmarks are generated from the feature points and the camera pose and landmarks are tracked. See <https://github.com/stella-cv/stella_vslam/blob/e6d8fa5d9593b153aad74a1504586aea38ccd2e1/src/stella_vslam/tracking_module.cc#L200>
